sonic_platform_base/sonic_xcvr/mem_maps/public/sff8636.py

```diff
@@ -50,7 +50,9 @@ def __init__(self, codes):
             ),
             CodeRegField(consts.CONNECTOR_FIELD, self.get_addr(0, 130), self.codes.CONNECTORS),
             RegGroupField(consts.SPEC_COMPLIANCE_FIELD, 
-                CodeRegField(consts.ETHERNET_10_40G_COMPLIANCE_FIELD, self.get_addr(0, 131), self.codes.ETHERNET_10_40G_COMPLIANCE),
+                CodeRegField(consts.ETHERNET_10_40G_COMPLIANCE_FIELD, self.get_addr(0, 131), self.codes.ETHERNET_10_40G_COMPLIANCE,
```

Comment:
@stephenxs what is this fixing?

Reply:
HI @prgeor 
This is to parse `10/40G/100G Ethernet Compliance Codes` in `Specification Compliance` (page 00H byte 131) in SFF8636. It uses enumerate to match the value. It works if there is only 1 bit set but can be broken if there is more than 1 bit set.
I believe this is inherited from SFF8434 in which the bit 7 is reserved and should be always zero. In that case, it work correctly.
But in SFF8636 bit 7 can be 1 indicating the `Extended Specification Compliance`. In that case, the logic using enumerate can fail.
By using mask the bit 7 is ignored in this paring and the logic works again.

Reply:
@stephenxs the extended compliance shown is correct:-

![image](https://github.com/sonic-net/sonic-platform-common/assets/45705344/6584b75d-c745-4f81-8656-ffc47671df98)


So, which CLI or functionality is impacted. For sure bit 7 in page0, 131 need to be taken care of but trying to understand the fix need to see whats the problem.

Reply:
@stephenxs also, as per the spec more the one bit is applicable for Fibre Channel (not for Ethernet)

![image](https://github.com/sonic-net/sonic-platform-common/assets/45705344/3755d4ac-38ab-452d-8afb-88ca24118b5c)


i don't see FC use case

Reply:
@prgeor the `10/40G Ethernet compliance code` was impacted. It is unknown in your example which is probably wrong.
can you try using ethtool to see the specification compliance?
I can also give you an example on my testbed
the `ethtool` output, transceiver codes is 0x88, `Transceiver type` has both 40G and 25G
```
root@sonic:/home/admin# ethtool -m sfp1
        Identifier                                : 0x11 (QSFP28)
        Extended identifier                       : 0x00
        Extended identifier description           : 1.5W max. Power consumption
        Extended identifier description           : No CDR in TX, No CDR in RX
        Extended identifier description           : High Power Class (> 3.5 W) not enabled
        Connector                                 : 0x23 (No separable connector)
        Transceiver codes                         : 0x88 0x00 0x00 0x00 0x00 0x00 0x00 0x00
        Transceiver type                          : 40G Ethernet: 40G Base-CR4
        Transceiver type                          : 25G Ethernet: 25G Base-CR CA-N
        Encoding                                  : 0x05 (64B/66B)
        BR, Nominal                               : 25500Mbps
        Rate identifier                           : 0x00
        Length (SMF,km)                           : 0km
        Length (OM3 50um)                         : 0m
        Length (OM2 50um)                         : 0m
        Length (OM1 62.5um)                       : 0m
        Length (Copper or Active cable)           : 3m
        Transmitter technology                    : 0xa0 (Copper cable unequalized)
        Attenuation at 2.5GHz                     : 4db
        Attenuation at 5.0GHz                     : 6db
        Attenuation at 7.0GHz                     : 8db
        Attenuation at 12.9GHz                    : 12db
        Vendor name                               : Mellanox
        Vendor OUI                                : 00:02:c9
        Vendor PN                                 : MCP1600-C003E26N
        Vendor rev                                : A5
        Vendor SN                                 : MT2001VS03432
        Date code                                 : 191227
        Revision Compliance                       : SFF-8636 Rev 2.0
        Module temperature                        : 0.00 degrees C / 32.00 degrees F
        Module voltage                            : 0.0000 V
```
but before change, `10/40G Ethernet Compliance Code` is `Unknown`
```
sfputil show eeprom -p Ethernet0
Ethernet0: SFP EEPROM detected
        Application Advertisement: N/A
        Connector: No separable connector
        Encoding: 64B/66B
        Extended Identifier: Power Class 1 Module (1.5W max.), No CLEI code present in Page 02h, No CDR in TX, No CDR in RX
        Extended RateSelect Compliance: Unknown
        Identifier: QSFP28 or later
        Length Cable Assembly(m): 3.0
        Nominal Bit Rate(100Mbs): 255
        Specification compliance:
                10/40G Ethernet Compliance Code: Unknown
                Extended Specification Compliance: 25GBASE-CR CA-25G-N or 50GBASE-CR2 with no FEC
                Fibre Channel Link Length: Unknown
                Fibre Channel Speed: Unknown
                Fibre Channel Transmission Media: Unknown
                Fibre Channel Transmitter Technology: Unknown
                Gigabit Ethernet Compliant Codes: Unknown
                SAS/SATA Compliance Codes: Unknown
                SONET Compliance Codes: Unknown
        Vendor Date Code(YYYY-MM-DD Lot): 2019-12-27   
        Vendor Name: Mellanox        
        Vendor OUI: 00-02-c9
        Vendor PN: MCP1600-C003E26N
        Vendor Rev: A5
        Vendor SN: MT2001VS03432   
```
after change, `10/40G Ethernet Compliance Code` is correct.
```
Ethernet0: SFP EEPROM detected
        Application Advertisement: N/A
        Connector: No separable connector
        Encoding: 64B/66B
        Extended Identifier: Power Class 1 Module (1.5W max.), No CLEI code present in Page 02h, No CDR in TX, No CDR in RX
        Extended RateSelect Compliance: Unknown
        Identifier: QSFP28 or later
        Length Cable Assembly(m): 3.0
        Nominal Bit Rate(100Mbs): 255
        Specification compliance:
                10/40G Ethernet Compliance Code: 40GBASE-CR4
                Extended Specification Compliance: 25GBASE-CR CA-25G-N or 50GBASE-CR2 with no FEC
                Fibre Channel Link Length: Unknown
                Fibre Channel Speed: Unknown
                Fibre Channel Transmission Media: Unknown
                Fibre Channel Transmitter Technology: Unknown
                Gigabit Ethernet Compliant Codes: Unknown
                SAS/SATA Compliance Codes: Unknown
                SONET Compliance Codes: Unknown
        Vendor Date Code(YYYY-MM-DD Lot): 2019-12-27   
        Vendor Name: Mellanox        
        Vendor OUI: 00-02-c9
        Vendor PN: MCP1600-C003E26N
        Vendor Rev: A5
        Vendor SN: MT2001VS03432   
```

Reply:
@stephenxs Why would spec won't call out Ethernet and refer to Fiber channel for multiple bits?

![image](https://github.com/sonic-net/sonic-platform-common/assets/45705344/d182bb58-14d1-49d8-bcc1-4751309bcb11)

Lets consider the output you mentioned where 10G/40G compliance shown as "40G CR4". If you see the extended specification the compliance is 25GBASE-CR or 50G BASE CR2. Does this mean this optics supports 40G speed as well as 100G speed? If the port is configured at 40G speed can this cable support 40G speed?

Reply:
@stephenxs the optics output you shared supports 1x50G, 1x25G, 1x100G speed. Please check.

Reply:
Hi @prgeor 
I think so according to the spec. Please refer the output of ethtool as well.